include/envoy/stats/scope.h

```diff
@@ -136,6 +138,31 @@ class Scope {
    */
   virtual Histogram& histogramFromString(const std::string& name, Histogram::Unit unit) PURE;
 
+  /**
+   * Creates a TextReadout from the stat name. Tag extraction will be performed on the name.
+   * @param name The name of the stat, obtained from the SymbolTable.
+   * @return a text readout within the scope's namespace.
+   */
+  TextReadout& textReadoutFromStatName(const StatName& name) {
+    return textReadoutFromStatNameWithTags(name, absl::nullopt);
+  }
+  /**
+   * Creates a TextReadout from the stat name and tags. If tags are not provided, tag extraction
+   * will be performed on the name.
+   * @param name The name of the stat, obtained from the SymbolTable.
+   * @param tags optionally specified tags.
+   * @return a text readout within the scope's namespace.
+   */
+  virtual TextReadout& textReadoutFromStatNameWithTags(const StatName& name,
+                                                       StatNameTagVectorOptConstRef tags) PURE;
+
+  /**
+   * TODO(#6667): this variant is deprecated: use textReadoutFromStatName.
```

Comment:
For new code I think we can remove this?

Reply:
We still need this for stats macros to work. However we should put the format-check at https://github.com/envoyproxy/envoy/blob/0ebcd1a5ea904e849b62763bcede94a5adea7c60/tools/code_format/check_format.py#L670 in this PR.